gadget compile doesn't print warnings if compilation works

## Description

When you compile a gadget with `ig make`, it only shows the warnings when there is an error too.

This is very confusing, as initially I thought the warnings were solved too, but it was until I made another error that I saw the warnings again.

I don't know much about eBPF, but usually warnings in C are kind of important and it's quite common to compile with `-Wall` or sometimes even `-Werror`.

```
$ IG="$PWD/../ig" make check_mount
Building check_mount
Pulling builder image ghcr.io/inspektor-gadget/ebpf-builder:main
main: Pulling from inspektor-gadget/ebpf-builder
Digest: sha256:849546bcee32e8329eb028dd25db1d3987a1c07da2a770451b5f2047ea901e4b
Status: Image is up to date for ghcr.io/inspektor-gadget/ebpf-builder:main
Build logs start:
clang -target bpf -Wall -g -O2 -I /work/include/ -D __TARGET_ARCH_x86 \
	-c /work/gadgets/check_mount/program.bpf.c -I /usr/include/gadget/amd64/ -o /out/amd64.bpf.o
clang -target bpf -Wall -g -O2 -I /work/include/ -D __TARGET_ARCH_arm64 \
	-c /work/gadgets/check_mount/program.bpf.c -I /usr/include/gadget/arm64/ -o /out/arm64.bpf.o
/work/gadgets/check_mount/program.bpf.c:84:9: warning: 'MS_BIND' macro redefined [-Wmacro-redefined]
   84 | #define MS_BIND 0x2000
      |         ^
/work/gadgets/check_mount/program.bpf.c:83:9: note: previous definition is here
   83 | #define MS_BIND 0x1000
      |         ^
/work/gadgets/check_mount/program.bpf.c:96:2: error: use of undeclared identifier 'asd'
   96 |         asd;
      |         ^
1 warning and 1 error generated.
make: *** [/out/Makefile.build:17: /out/amd64.bpf.o] Error 1
make: *** Waiting for unfinished jobs....
/work/gadgets/check_mount/program.bpf.c:84:9: warning: 'MS_BIND' macro redefined [-Wmacro-redefined]
   84 | #define MS_BIND 0x2000
      |         ^
/work/gadgets/check_mount/program.bpf.c:83:9: note: previous definition is here
   83 | #define MS_BIND 0x1000
      |         ^
/work/gadgets/check_mount/program.bpf.c:96:2: error: use of undeclared identifier 'asd'
   96 |         asd;
      |         ^
1 warning and 1 error generated.
make: *** [/out/Makefile.build:17: /out/arm64.bpf.o] Error 1
Build logs end
Error: builder container exited with status 2
make: *** [Makefile:70: check_mount] Error 1
```

[ 1 paragraph concisely describing the bug ]

## Impact

Don't accidentally have bugs in gadgets that you didn't see because warnings are not printed.

## Expected behavior

Compilation warnings of a gadget are shown

--- a/cmd/common/image/build.go
+++ b/cmd/common/image/build.go
@@ -474,17 +474,20 @@ func buildInContainer(opts *cmdOpts, conf *buildFile) error {
 	case status = <-statusCh:
 	}
 
-	if status.StatusCode != 0 || common.Verbose {
-		opts := container.LogsOptions{ShowStdout: true, ShowStderr: true}
-		out, err := cli.ContainerLogs(ctx, resp.ID, opts)
-		if err != nil {
-			return fmt.Errorf("getting builder container logs: %w", err)
-		}
+	outputOpts := container.LogsOptions{ShowStderr: true}
 
-		fmt.Println("Build logs start:")
-		stdcopy.StdCopy(os.Stdout, os.Stderr, out)
-		fmt.Println("Build logs end")
+	if status.StatusCode != 0 || common.Verbose {
+		outputOpts.ShowStdout = true
 	}
+
+	out, err := cli.ContainerLogs(ctx, resp.ID, outputOpts)
+	if err != nil {
+		return fmt.Errorf("getting builder container logs: %w", err)
+	}
+
+	fmt.Println("Build logs start:")
+	stdcopy.StdCopy(os.Stdout, os.Stderr, out)
+	fmt.Println("Build logs end")
 
 	if status.StatusCode != 0 {
 		return fmt.Errorf("builder container exited with status %d", status.StatusCode)

